### ISSUE TITLE
Databricks: CREATE TEMP VIEW does not parse, but CREATE TEMPORARY VIEW does

### ISSUE THREAD

**Issue by cchambers-rdi** [sqlfluff/sqlfluff#7948]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following Databricks SQL statement errors out when parsing

``` sql
CREATE TEMP VIEW v AS
SELECT 1 AS x
```

However, it parses fine if I use the sparksql dialect.

The following parses fine in the Databricks dialect:

``` sql
CREATE TEMPORARY VIEW v AS
SELECT 1 AS x
```

### Expected Behaviour

The statement:

``` sql
CREATE TEMP VIEW v AS
SELECT 1 AS x
```

Should parse as expected.

### Observed Behaviour

I get a parse error.

```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE TEMP VIEW
                       | v AS\nSELECT 1 AS x'
WARNING: Parsing errors found and dialect is set to 'databricks'. Have you configured your dialect correctly?
All Finished!
```

### How to reproduce

Create a SQL file with the contents:

``` sql
CREATE TEMP VIEW v AS
SELECT 1 AS x
```

Parse it with sqlfluff
`sqlfluff lint file.sql --dialect databricks`

### Dialect

Databricks

### Version

4.2.0

### Configuration

[tool.sqlfluff.core]
dialect = "databricks"
templater = "python"
sql_file_exts = ".sql"
output_line_length = 88
max_line_length = 88
max_parse_depth = 500
ignore_paths = ["docs/**"]

[tool.sqlfluff.templater.python.context]
bronze_catalog = "bronze_catalog"
working_catalog = "working_catalog"
catalog = "catalog"
source_catalog = "source_catalog"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
